### ISSUE TITLE
Easier recovery from temporarily missing data files ("No data found!")

### ISSUE THREAD

**Issue by victorhooi** [transmission/transmission#83]
I often download torrents via Transmission to my network drive (CIFS or AFP).

However, if I restart my Macbook and the drive isn't mounted yet - Transmission will stop that torrent (expected), but will not allow me to resume it (even if I later mount the network drive) without first removing the torrent then re-adding it. The message given below the torrent is:

> Error: No data found! Ensure your drives are connected or use "Move Data File To...". To re-download, remove the torrent and re-add it.

Is there any way of making the resume process easier, so that you can simply right-click and resume and it will check if the files were where they were before?

**Comment by andreygursky** [comment]
Sometimes I move torrent data (while transmission-daemon is not running), then it complains as you described. But in this case it is enough to issue a "move" command to the new destination, then torrent can be started without re-adding. This would be a nice workaround for you: dummy "move" to the same place, then start (if it would work). In case you're missing many torrents, since the whole drive wasn't mounted, I believe the best solution is to restart transmission. If you're facing this issue often, did you think about to remove it from autostart (or ensure it starts only after successful network drive mounts)? Re-adding is not a workaround at all IMHO, since torrents must be rechecked (however if they're not on ZFS, it might be not so bad :) on the other hand bittorrent can be treated as backup itself).

**Comment by CloCkWeRX** [comment]
I have a similar issue with extra HDD - if its not mounted but a torrent starts, it gets into an errored state.

While its only a few clicks to copy magnet link to clipboard and open url (3?); it would be a *lot* better if there was "Ignore missing data" checkbox in Set Location or something like that to allow an easier restart.

**Comment by victorhooi** [comment]
Is there any update on this, or how hard it would be to fix? =).

Thanks!

**Comment by cfpp2p** [comment]
[TRAC-4537](https://trac.transmissionbt.com/ticket/4537#comment:4)

**Comment by gaieges** [comment]
Here's a quick little script to re-add busted magnets:

```bash
#!/bin/bash

TRANSMISSION_ARGS="-n username:password"

START_ID=23
END_ID=38



for i in $(seq $START_ID $END_ID); do

  info=$(transmission-remote $TRANSMISSION_ARGS -t $i -i)
  magnet=$(echo "$info" | grep magnet | sed 's/Magnet: //')

  transmission-remote $TRANSMISSION_ARGS -t $i --remove-and-delete
  transmission-remote $TRANSMISSION_ARGS -a $magnet

done
```

**Comment by Fitap** [comment]
Here another user with **Error: No data found! Ensure your drives are connected or use "Set Location". To re-download, remove the torrent and re-add it.**

I run gentoo and I already have my point mounted ok.

**Comment by timshannon** [comment]
I have a similar issue.  I recently lost a drive, and while I backup all of my important files, I don't backup my torrented files.  Transmission says to delete and re-add the torrent, but I don't have the original torrent files.  It would be nice to just have an option to automatically start the torrent from scratch, and preserve the existing download locations and other settings.

**Comment by lesmana** [comment]
just for the record: you can make transmission restart a torrent by deleting the "resume" files.

    rm ~/.config/transmission/resume/torrentyouwanttorestart.resume

did not test much. works at least for the case where i deliberately deleted the partially downloaded files from command line and just wanted transmission to restart the torrent from scratch.

**Comment by eyalroz** [comment]
@lesmana : Well, the UI doesn't tell us that... Also, do you need to restart the app for this to work?

Anyway, for me, I'd like this feature for torrents with multiple independently-usable files, so that I can delete files I'm done with, change the selected files to download, and resume.

**Comment by vasecthomas** [comment]
> just for the record: you can make transmission restart a torrent by deleting the "resume" files.
> 
> ```
> rm ~/.config/transmission/resume/torrentyouwanttorestart.resume
> ```
> 
> did not test much. works at least for the case where i deliberately deleted the partially downloaded files from command line and just wanted transmission to restart the torrent from scratch.

this also deletes the reference where the files were stored, so you will be clicking around again to save everything in the appropriate folders

**Comment by j-cr** [comment]
So, seems like the patch that fixes this issue [was provided 8 years ago](https://trac.transmissionbt.com/attachment/ticket/4537/no-data-found-override.patch), is there something wrong with it or does something else block this issue from being resolved?

**Comment by VarunAgw** [comment]
> just for the record: you can make transmission restart a torrent by deleting the "resume" files.
> 
> ```
> rm ~/.config/transmission/resume/torrentyouwanttorestart.resume
> ```
> 
> did not test much. works at least for the case where i deliberately deleted the partially downloaded files from command line and just wanted transmission to restart the torrent from scratch.

Thanks. Very helpful. Just a note in Ubuntu, the path is:

`/var/lib/transmission-daemon/info/resume/`

**Comment by eyalroz** [comment]
@VarunAgw : Note that this feature request is about a proper _resumption_, not a restart.

**Comment by VarunAgw** [comment]
@eyalroz Oh, my bad.

**Comment by jordanade** [comment]
So I guess using "Set data location" really does work (I think?), but the UX on my client doesn't show that it is in a normal state. It continues to show an error, but then sometimes suddenly begins working. This functionality needs work!

**Comment by GregoryGH** [comment]
Really annoying. Please recreate files on Force Start f.e.

**Comment by voarsh2** [comment]
Area has some work needed.

**Comment by mohkale** [comment]
Still an issue, and a really annoying one for me since I download to multiple different hard drives and sometimes won't have one drive mounted whilst I'm trying to download something to another drive. The option is either to backup the whole .config/transmission directoriy or YOLO it and just live with transmission complaining about missing data until I manually re-add all the torrents that were corrupted with their updated locations. I could probably write a script to automate this, but even then I'll have to run a verify-torrent command on each torrent just because transmission won't pick up the download path again, even if nothing in the underlying files content has changed and therefore most of the data would be identicle.

**Comment by dchmelik** [comment]
Let's be clear that this could be from not just temporarily, but permanently, missing data files, so there should be an option to restart downloading from zero anyway; a common thing is one runs out of space, deletes stuff, re-downloads later, which I and many people I know have been doing since the 1990s, and people who taught us have done since when computers were still an obscure/new/nerd thing.  It's common I download & seed large numbers of things that are free (public domain, Free/Libre/Opensource Software (FLS, OSS, FOSS, FLOSS) including POSIX-based (UNIX, GNU, Linux) operating systems (OS), even alternative/old versions for other hardware and still-used old software) until I run out of space, but then after I either delete things I'm not using, or upgrade storage drives, I have to remove & reopen perhaps hundreds torrents when the data files aren't going to be there in the first place... ends up taking hours/days of reopening without an option to simply restart from zero.  Another reason is it's also common to have data accidents/losses which results in same problem.
        I know that it should be an option--not automatic--to restart from zero, because the reason I guess it doesn't is (as others describe above) is it may fill up the drive because data may have been moved elsewhere then duplicate... just pointing out that isn't always the case as many cases also exists deletion was 'permanent' (until restarting which should never require remove & reload, because ruins 'sort by age').

**Comment by 07416** [comment]
> just for the record: you can make transmission restart a torrent by deleting the "resume" files.
> 
> ```
> rm ~/.config/transmission/resume/torrentyouwanttorestart.resume
> ```
> 

Is it safe to remove **all** `.resume` files, while the client is stopped? Hundreds of my torrents show the error, non-feasible to filter manually.

**Comment by aki-k** [comment]
@07416 Fixed it with this bash alias:

alias t-cli="rm -f ~/.config/transmission/resume/*; transmission-cli ${@}"

**Comment by Pentaphon** [comment]
Somebody posted this info in a duplicate that was closed:

>I have found what is causing this issue. It's not a bug, rather just bad design. When specifying a directory transmission looks for the file inside that directory. Essentially you need to traverse one level higher than what you would expect. For example, If I wanted to specify the data location to be:
/home/github/folder
I would need to select /home/github as my file location, not /home/github/folder, as transmission appends /folder to the directory specified. If I had selected /home/github/folder as my file location, it would be looking for /home/github/folder/folder resulting in the error 'no data found'

https://github.com/transmission/transmission/issues/345#issuecomment-444883740

**Comment by neaumusic** [comment]
I need the ability to re-link files, I downloaded the torrent to my Downloads folder but moved it to a more permanent Dropbox folder, and it's important that I seed these files

**Comment by tearfur** [comment]
This issue has diverged to a lot of different requests all with their own niche, difficult to account for them all in the UI.

Anyway, I want to try address the major concerns here:

1. *The data files were temporarily gone, but now they are accessible again on the same file path. I want an easy way to resume the torrent.*
   
   Just resume the torrent, the error will clear. @victorhooi @CloCkWeRX @Fitap

**Comment by tearfur** [comment]
Adding the `feature disparity` label since this feature request is now resolved only in the WebUI.

Ideally I'd like some user feedback to how intuitive the new `Verify Local Data (force)` context menu item is before considering porting the feature to other clients, but PRs are always welcome.

**Comment by Pentaphon** [comment]
@tearfur did you see [my comment here.](https://github.com/transmission/transmission/issues/83#issuecomment-1588477589) I think that quote was getting to the root cause of the issue. I'm wondering what your thoughts are on those observations.

**Comment by tearfur** [comment]
That very beginning of this issue was about how the user found it very inconvenient to have to remove and re-add a torrent to resume it.

I'd consider your comment an entirely separete problem.

**Comment by Pentaphon** [comment]
> I'd consider your comment an entirely separete problem.

@tearfur That's the thing. I don't think anybody has opened an issue for that original "no data found" bug and this is the only issue on github that refers to the original "no data found" bug. [All "no data found" issues are considered a duplicate of this very issue and are closed.](https://github.com/transmission/transmission/issues/345#issuecomment-1022840615) I'd open a new issue that seeks to treat the "no data found" issue as a separate problem but it will just be closed. I'd like to address the root cause of the issue rather than just recovering from it when it happens. 

What do you think should be done about this?

@ckerr I am CC'ing you so that you are aware of this dilemma of how to create an issue that seeks to solve the root cause of "no data found" bug which is being referred to as "a design flaw".
